dodiscover/constraint/skeleton.py

```diff
@@ -0,0 +1,470 @@
+        - 'neighbors': This only conditions on adjacent variables to that of 'x_var' and 'y_var'.
+          This refers to the traditional PC algorithm :footcite:`Meek1995`
+        - 'neighbors_path': This is 'neighbors', but restricts to variables with an adjacency path
+          from 'x_var' to 'y_var'. This is a variant from the RFCI paper :footcite:`Colombo2012`
+    """
+
+    adj_graph_: nx.Graph
+    sep_set_: Dict[str, Dict[str, List[Set[Any]]]]
+    remove_edges: Set
+
+    def __init__(
+        self,
+        ci_estimator: BaseConditionalIndependenceTest,
+        sep_set: Dict[str, Dict[str, List[Set[Any]]]] = None,
+        alpha: float = 0.05,
+        min_cond_set_size: int = 0,
+        max_cond_set_size: int = None,
+        max_combinations: int = None,
+        skeleton_method: str = "neighbors",
+        keep_sorted: bool = False,
+        **ci_estimator_kwargs,
+    ) -> None:
+        self.ci_estimator = ci_estimator
+        self.sep_set = sep_set
+        self.alpha = alpha
+        self.ci_estimator_kwargs = ci_estimator_kwargs
+        self.skeleton_method = skeleton_method
+
+        # control of the conditioning set
+        self.min_cond_set_size = min_cond_set_size
+        self.max_cond_set_size = max_cond_set_size
+        self.max_combinations = max_combinations
+
+        # for tracking strength of dependencies
+        self.keep_sorted = keep_sorted
+
+    def _initialize_params(self):
+        """Initialize parameters for learning skeleton.
+
+        Parameters
+        ----------
+        nodes : list of nodes
+            The list of nodes that will be present in the learned skeleton graph.
+        """
+        # error checks of passed in arguments
+        if self.max_combinations is not None and self.max_combinations <= 0:
+            raise RuntimeError(f"Max combinations must be at least 1, not {self.max_combinations}")
+
+        if self.skeleton_method not in ACCEPTED_MARKOVIAN_SKELETON_METHODS:
+            raise ValueError(
+                f"Skeleton method must be one of {ACCEPTED_MARKOVIAN_SKELETON_METHODS}, "
+                f"not {self.skeleton_method}."
+            )
+
+        if self.sep_set is None:
+            # keep track of separating sets
+            self.sep_set_ = defaultdict(lambda: defaultdict(list))
+        else:
+            self.sep_set_ = self.sep_set
+
+        # control of the conditioning set
+        if self.max_cond_set_size is None:
+            self.max_cond_set_size_ = np.inf
+        else:
+            self.max_cond_set_size_ = self.max_cond_set_size
+        if self.min_cond_set_size is None:
+            self.min_cond_set_size_ = 0
+        else:
+            self.min_cond_set_size_ = self.min_cond_set_size
+        if self.max_combinations is None:
+            self.max_combinations_ = np.inf
+        else:
+            self.max_combinations_ = self.max_combinations
+
+    def fit(self, context: Context):
```

Comment:
This only returns the skeleton, but does not direct edges when possible (using the 4 rules, such as colliders)? Maybe it makes sense to add a function to direct the edges as well based on the separation sets.